docs/actions.md

```diff
@@ -1,182 +1,129 @@
 # Actions
 
-Use [actions](/docs/api.md#actions) to update the state.
+Actions are functions which take the current [state](/docs/state.md) and return a partial state or a [thunk](#thunks). Actions represent the intent to manipulate the state tree.
 
-```jsx
-app({
-  state: "Hi.",
-  view: (state, actions) =>
-    <h1 onclick={actions.ucase}>
-      {state}
-    </h1>,
-  actions: {
-    ucase: state => state.toUpperCase()
-  }
-})
-```
-
-An action must return a partial state or a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that resolves to a partial state. See [Side Effects](#side-effects).
+[Try it Online](https://codepen.io/hyperapp/pen/qRMEGX?editors=0010)
 
 ```jsx
 app({
   state: {
-    count: 0,
-    maxCount: 10
+    text: "Hello!",
+    defaultText: "Nice weather today."
   },
-  view: (state, actions) =>
-    <main>
+  view: (state, { setText }) =>
+    <div>
       <h1>
-        {state.count}
+        {state.text.trim() === ""
+          ? state.defaultText
+          : state.text}
       </h1>
-      <button onclick={actions.up}>+</button>
-    </main>,
+      <input
+        autofocus
+        value={state.text}
+        oninput={e => setText(e.target.value)}
+      />
+    </div>,
   actions: {
-    up: ({ count, maxCount }) => ({
-      count: count + (maxCount > count ? 1 : -maxCount)
-    })
+    setText: (state, actions, text) => ({ text })
   }
 })
 ```
 
-You can pass data to actions as well.
-
-```jsx
-app({
-  state: {
-    count: 0
-  },
-  view: (state, actions) =>
-    <main>
-      <h1>
-        {state.count}
-      </h1>
-      <button onclick={() => actions.up(1)}>+</button>
-    </main>,
-  actions: {
-    up: ({ count }, actions, data = 0) => ({
-      count: count + data
-    })
-  }
-})
-```
+Actions are often called as a result of user events triggered on the [view](/docs/view.md) or inside application [events](/docs/events.md).
 
-## Side Effects
+## Thunks
 
-Actions are not required to have a return value. You can use them to call other actions, for example after an async operation has completed.
+Actions can return a function instead of a partial state. This function is called a _thunk_. They operate like regular actions but will not trigger a state update unless you call [`update`](/docs/api.md#update) from within the thunk function.
```

Comment:
unless [`update`](/docs/api.md#update) is called from within the thunk function.